Windows CMake builds are failing for paqa_dither with shared libraries

Windows CMake builds are failing with:
```
C:/mingw64/bin/../lib/gcc/x86_64-w64-mingw32/14.2.0/../../../../x86_64-w64-mingw32/bin/ld.exe: D:/a/portaudio/portaudio/qa/paqa_dither.c:164:(.text+0xb2f): undefined reference to `PaUtil_Generate16BitTriangularDither'
```

The paqa_dither test uses internal functions like PaUtil_InitializeTriangularDitherState () and PaUtil_Generate16BitTriangularDither(). The autoconf build has special targets to handle that.
But the CMake build fails on Windows when using shared (dynamic) libraries. (Thank you CI.)

_Originally posted by @philburk in https://github.com/PortAudio/portaudio/issues/1085#issuecomment-3413423064_

--- a/qa/CMakeLists.txt
+++ b/qa/CMakeLists.txt
@@ -1,6 +1,8 @@
 add_test(paqa_errs)
 add_test(paqa_devs)
-add_test(paqa_dither)
+if(LINK_PRIVATE_SYMBOLS)
+  add_test(paqa_dither)
+endif()
 add_test(paqa_latency)
 
 subdirs(loopback)

